File: src/leetcode/java/MaximumDepthOfBinaryTree_104.java
package leetcode.java;

public class MaximumDepthOfBinaryTree_104 {
  // 1. if root == null, return 0;
  // 2. else call maxDepth recursive.
  public int maxDepth(TreeNode root) {
    if (root == null) return 0;
    return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
  }
  
  class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
      val = x;
    }
  }
}

File: src/leetcode/python/MaximumDepthOfBinaryTree_104.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class MaximumDepthOfBinaryTree_104(object):
    def maxDepth(self, root):
        """
        1. if root == null, return 0;
        2. else call maxDepth recursive.
        """
        if (root == None): return 0
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
